llvm/lib/Target/AIE/AIE2TargetTransformInfo.h

```diff
@@ -54,10 +54,10 @@ class AIE2TTIImpl : public BasicTTIImplBase<AIE2TTIImpl> {
   void getUnrollingPreferences(Loop *L, ScalarEvolution &SE,
                                TTI::UnrollingPreferences &UP,
                                OptimizationRemarkEmitter *ORE) {
-    UP.Partial = UP.Runtime = true;
-    UP.AllowExpensiveTripCount = true;
     UP.Threshold = 200;
     BaseT::getUnrollingPreferences(L, SE, UP, ORE);
+    UP.Partial = UP.Runtime = false;
```

Comment:
This is nice! Now our decisions are not changed by BaseT implementation! Maybe those small changes can be merged in small PRs.